RecoLocalTracker/SiStripClusterizer/plugins/ClustersFromRawProducer.cc

```diff
@@ -308,10 +317,16 @@ try {
     
 
     // If Fed hasnt already been initialised, extract data and initialise
-    if (!done[fedId]) { buffers[fedId].reset(fillBuffer(fedId, rawColl)); done[fedId]=true;}
-    auto buffer = buffers[fedId].get();
-    if unlikely(!buffer) continue;
-    
+    sistrip::FEDBuffer * buffer = done[fedId];
+    if (!buffer) { 
+      buffer = fillBuffer(fedId, rawColl).release();
+      if (!buffer) { continue;}
+      sistrip::FEDBuffer * exp = nullptr;
+      if (done[fedId].compare_exchange_strong(exp, buffer)) buffers[fedId].reset(buffer);
+      else { delete buffer; buffer = done[fedId]; }
```

Comment:
You could also have done ```buffer = exp``` since after a failed ```compare_exchange_strong```, ```exp``` contains the value that was in the ```atomic```.